### PR TITLE
Don't use deny list in driver

### PR DESCRIPTION
**Pull request by vkgnosis**
The deny list is only needed for the order book. To avoid confusion
let's not use it in the driver at all.

This came up on the staging config update to clear the deny listed tokens.

### Test Plan
existing tests, shouldn't affect anything

### CODE REVIEW

File: solver/src/main.rs [hunk 1]
```diff
@@ -200,12 +200,11 @@ async fn main() {
 
     let pool_aggregator = PoolAggregator::from_providers(&pair_providers, &web3).await;
 
-    // TODO - use Filtered-Cached PoolFetchers here too.
```
Comment (bh2smith) [inline solver/src/main.rs:203]: Are we removing this because its no longer necessary? Or were you going to introduce it here?

**Review by bh2smith — APPROVED** [review]

File: solver/src/main.rs (hunk 1 above)
Comment (vkgnosis) [reply solver/src/main.rs:203]: As I am saying in the commit message I don't think the filtering part ever made sense. But I guess we might still want to cache this. I'll add that part of the comment back.

File: solver/src/main.rs [hunk 2]
```diff
@@ -200,12 +200,12 @@ async fn main() {
...
+    // TODO: use caching pool fetcher
     let price_estimator = Arc::new(BaselinePriceEstimator::new(
         Box::new(pool_aggregator),
         gas_price_estimator.clone(),
         base_tokens.clone(),
-        Arc::new(ListBasedDetector::deny_list(args.shared.unsupported_tokens)),
```
Comment (nlordell) [inline solver/src/main.rs:208]: Do we use `unsupported_tokens` in other places in the driver? If not, we should probably move it from the `shared::Options` to be `orderbook` specific.

Comment (vkgnosis) [reply solver/src/main.rs:208]: It's not used so I will add that to this PR.

**Review by nlordell — APPROVED** [review]

File: solver/src/main.rs (hunk 2 above)
Comment (fleupold) [reply solver/src/main.rs:208]: Maybe also add a comment why it's fine to not filter additional tokens (the orderbook does the filtering for us)?

**Review by fleupold — APPROVED** [review]

**Pull request merged**